language/move-prover/abigen/src/abigen.rs

```diff
@@ -123,7 +159,23 @@ impl<'env> Abigen<'env> {
                 },
             )
             .collect::<anyhow::Result<_>>()?;
-        Ok(ScriptABI::new(name, doc, code, ty_args, args))
+
+        // This is a transaction script, so include the code, but no module ID
+        if module_env.is_script_module() {
+            let code = self.load_compiled_bytes(&module_env)?.to_vec();
+            Ok(ScriptABI::TransactionScript(TransactionScriptABI::new(
+                name, doc, code, ty_args, args,
+            )))
+        } else {
+            // This is a script function, so no code. But we need to include the module ID
+            Ok(ScriptABI::ScriptFunction(ScriptFunctionABI::new(
```

Comment:
No idea what to do, but not every script function is invokable as a script transaction/from execute_script_function. Does that matter for the ABIs?

Reply:
Thanks so much for point this out. Updated the PR to handle this, and added a test case to make sure that we don't generate ABIs for non-callable script functions.

Reply:
Looks good to me!